python/amici/pandas.py

```diff
@@ -367,28 +367,27 @@ def _get_names_or_ids(model, variable, by_id):
     # check whether variable type permitted
     variable_options = ['Parameter', 'FixedParameter', 'Observable', 'State']
     if variable not in variable_options:
-        raise ValueError('variable must be in ' + str(variable_options))
+        raise ValueError('Variable must be in ' + str(variable_options))
 
-    # functions to extract attributes
-    namegetter = getattr(model, 'get' + variable + 'Names')
-    idgetter = getattr(model, 'get' + variable + 'Ids')
+    # extract attributes
+    names = list(getattr(model, f'get{variable}Names')())
+    ids = list(getattr(model, f'get{variable}Ids')())
 
     # extract labels
-    if not by_id and len(set(namegetter())) == len(namegetter()) \
-            and getattr(model, f"has{variable}Names")():
+    if not by_id and len(set(names)) == len(names):
```

Comment:
should check for `has{variable}Names` here, otherwise names = [] will lead to false positivies here.

Reply:
I assumed that has{variable}Names() == len(get{variable}Names()). That is not true, then? What is the exact intended behavior? If that is not done yet, I would also document it.

Reply:
> boolean indicating whether parameter names were set

If len(getParameterNames) is zero, we do not know whether (i) the model was compiled without overriding getParameterNames (ii) np == 0. So if len(getParameterNames) == 0, you would have to explicitly check if np > 0 to see whether Names are _undefined_. Apparently we screwed up the implementation as the right thing to check is `np == 0 || getParameterNames.size()`

Reply:
ok, now it makes sense, thanks! I will change that then.